genic/thermal.c

```diff
@@ -76,33 +74,50 @@ init_thermalvel(const double v_amp, double max_fd,const double min_fd)
 
     gsl_integration_workspace_free (w);
 
-    double total_frac = fermi_dirac_cumprob[LENGTH_FERMI_DIRAC_TABLE-1]/total_fd;
+    double total_frac = thermals->fermi_dirac_cumprob[LENGTH_FERMI_DIRAC_TABLE-1]/total_fd;
     //Normalise total integral to unity
     for(i = 0; i < LENGTH_FERMI_DIRAC_TABLE; i++)
-        fermi_dirac_cumprob[i] /= fermi_dirac_cumprob[LENGTH_FERMI_DIRAC_TABLE - 1];
+        thermals->fermi_dirac_cumprob[i] /= thermals->fermi_dirac_cumprob[LENGTH_FERMI_DIRAC_TABLE - 1];
 
     /*Initialise the GSL table*/
-    fd_intp = gsl_interp_alloc(gsl_interp_cspline,LENGTH_FERMI_DIRAC_TABLE);
-    fd_intp_acc = gsl_interp_accel_alloc();
-    gsl_interp_init(fd_intp,fermi_dirac_cumprob, fermi_dirac_vel,LENGTH_FERMI_DIRAC_TABLE);
+    thermals->fd_intp = gsl_interp_alloc(gsl_interp_cspline,LENGTH_FERMI_DIRAC_TABLE);
+    thermals->fd_intp_acc = gsl_interp_accel_alloc();
+    gsl_interp_init(thermals->fd_intp,thermals->fermi_dirac_cumprob, thermals->fermi_dirac_vel,LENGTH_FERMI_DIRAC_TABLE);
     return total_frac;
 }
 
-/*Add a randomly generated thermal speed in v_amp*(min_fd, max_fd) to a 3-velocity*/
+/*Generate a table of random seeds, one for each pencil.*/
+unsigned int *
+init_rng(int Seed, int Nmesh)
```

Comment:
The memory consumption goes like N**3 -- and will be too much for 4096. With the pfft `local_ni` and `local_i_start` We can probably work something out later.

Reply:
It goes like N^2